### PR TITLE
Implement benchmark for secondary index retrieval speeds

### PR DESCRIPTION
**Pull request by orkunkl**
This PR implements the benchmark mentioned in https://github.com/iov-one/weave/issues/1059:
- Secondary index retrieval speeds in the range of 1-50000 element size.

### CODE REVIEW

**Comment by orkunkl** [comment]
```     | goos: darwin
     | goarch: amd64
     | pkg: github.com/iov-one/weave/benchmark
     | BenchmarkSecondaryIndex/index_length_2_amount_1-4         	  120460	      9211 ns/op
     | BenchmarkSecondaryIndex/index_length_2_amount_10-4        	  138662	      9117 ns/op
     | BenchmarkSecondaryIndex/index_length_2_amount_100-4       	  126034	      9422 ns/op
     | BenchmarkSecondaryIndex/index_length_2_amount_1000-4      	  122908	      9491 ns/op
     | BenchmarkSecondaryIndex/index_length_2_amount_10000-4     	  120938	      9381 ns/op
     | BenchmarkSecondaryIndex/index_length_2_amount_50000-4     	  126351	      9145 ns/op
     | BenchmarkSecondaryIndex/index_length_5_amount_1-4         	  134464	      9752 ns/op
     | BenchmarkSecondaryIndex/index_length_5_amount_10-4        	  115273	      9528 ns/op
     | BenchmarkSecondaryIndex/index_length_5_amount_100-4       	  133078	      9728 ns/op
     | BenchmarkSecondaryIndex/index_length_5_amount_1000-4      	  124942	      9538 ns/op
     | BenchmarkSecondaryIndex/index_length_5_amount_10000-4     	  125848	      9986 ns/op
     | BenchmarkSecondaryIndex/index_length_5_amount_50000-4     	  127486	      9650 ns/op
[32mPASS | github.com/iov-one/weave/benchmark 15.587s[0m
```

**Comment by orkunkl** [comment]
@husio I am not sure if executing the bucket population inside of the benchmark execution dilutes the precision of the benchmark?

**Comment by iov-bot-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/iov-one/weave/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: orm/benchmarks_test.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	}
+
+	for _, bm := range benchmarks {
+		b.Run(bm.name, func(b *testing.B) {
+			for i := 0; i < b.N; i++ {
+				var db weave.KVStore
+				if bm.isInMem {
+					db = store.MemStore()
+				} else {
+					db, err = bnsd.CommitKVStore(dbPath)
```
Comment (orkunkl) [inline orm/benchmarks_test.go]: It gives an error here. `weave.CommitKVStore` does not inherit(or what to call in go) `weave.KVStore`. What should I do?

**Pull request closed without merging**